File: Subarray Sum Equals K.js
var subarraySum = function (nums, k) {
  let map = new Map();
  map.set(0, 1);
  let res = 0;

  let sum = 0;
  for (let i = 0; i < nums.length; i++) {
    sum += nums[i];
    let diff = sum - k;

    if (map.has(diff)) {
        console.log(map.get(diff))
      res += map.get(diff);

    }

    if (map.has(sum)) {
      map.set(sum, map.get(sum) + 1);
    } else {
      map.set(sum, 1);
    }
  }
  return res;
};

File: subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        res = 0
        curSum = 0
        prefixSums = Counter([0])
        
        for n in nums:
            curSum += n
            diff = curSum - k
            
            res += prefixSums.get(diff,0)
            prefixSums[curSum] += 1 
        return res